pkg/authorization/static/static.go

```diff
@@ -60,17 +60,33 @@ func (saConfig StaticAuthorizationConfig) Matches(a authorizer.Attributes) bool
 	isAllowed := func(staticConf string, requestVal string) bool {
 		if staticConf == "" {
 			return true
-		} else {
-			return staticConf == requestVal
 		}
+		return staticConf == requestVal
+	}
+	isGroupAllowed := func(configGroups, requestGroups []string) bool {
+		if len(configGroups) == 0 {
+			return true
+		}
+		// O(n^2) is fine here as the groups are small. Optimize if n grows large.
+		for _, configGroup := range configGroups {
+			for _, requestGroup := range requestGroups {
+				if configGroup == requestGroup {
+					return true
+				}
+			}
+		}
+		return false
 	}
 
 	userName := ""
+	userGroups := []string{}
 	if a.GetUser() != nil {
 		userName = a.GetUser().GetName()
+		userGroups = a.GetUser().GetGroups()
 	}
 
 	if isAllowed(saConfig.User.Name, userName) &&
+		isGroupAllowed(saConfig.User.Groups, userGroups) &&
```

Comment:
I am not sure I follow the use case. My expectation: "I want all users from group X to be able to proceed". However the implementation seems to require the user to be in certain groups beside it matching the username.